app/src/main/java/mok/it/app/mokapp/feature/project_add_participants/AddOrRemoveMembersFragment.kt

```diff
@@ -0,0 +1,217 @@
+        AddParticipantsViewModelFactory(args.projectId)
+    }
+
+    enum class DialogType {
+        NONE, CONFIRM
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View = ComposeView(requireContext()).apply {
+        setContent {
+            AddParticipantsScreen(viewModel)
+        }
+    }
+
+    @Composable
+    fun AddParticipantsScreen(viewModel: AddOrRemoveParticipantsViewModel) {
+        val uiState by viewModel.uiState.collectAsState()
+
+        val chipState =
+            rememberChipTextFieldState<Chip>() // this is not saved at configuration change yet
+        var searchQuery by rememberSaveable { mutableStateOf("") }
+        val filteredUnselectedUsers = getFilteredUsers(
+            searchQuery,
+            chipState,
+            uiState.selectedUserIds
+        ).filter { !uiState.selectedUserIds.contains(it.documentId) }
+        val users = viewModel.users.observeAsState(initial = emptyList()).value
+
+        var showDialog by rememberSaveable { mutableStateOf(DialogType.NONE) }
+
+        Scaffold(
+            bottomBar = {
+                Button(modifier = Modifier
+                    .padding(8.dp)
+                    .fillMaxWidth(),
+                    enabled = uiState.selectedUsersChanged,
+                    onClick = {
+                        showDialog = DialogType.CONFIRM
+                    }) {
+                    Text(text = "Mentés")
+                }
+            },
+        ) { padding ->
+            Column(
+                modifier = Modifier.padding(bottom = padding.calculateBottomPadding())
+            ) {
+                when (showDialog) {
+                    DialogType.CONFIRM -> {
+                        OkCancelDialog(
+                            text = "Biztos, hogy el szeretnéd menteni a módosításokat? Ha valakit törölsz a projektből, az elveszíti az összes eddig megszerzett mancsát rajta.",
+                            onConfirm = {
+                                viewModel.updateMembersOfProject()
+                                Toast.makeText(
+                                    context,
+                                    "Résztvevők listája módosítva!",
+                                    Toast.LENGTH_SHORT
+                                )
+                                    .show()
+                                showDialog = DialogType.NONE
+                            },
+                            onDismiss = {
+                                showDialog = DialogType.NONE
+                            },
+                        )
+                    }
+
+                    DialogType.NONE -> {}
+                }
+                SearchField(
+                    searchQuery = searchQuery,
+                    chipState = chipState,
+                    onValueChange = { searchQuery = it },
+                )
+
+                LazyColumn {// selected users
```

Comment:
Ez itt nem lesz jó. Mivel ez egy LazyColumn, ezért végtelen sok függőleges helyet elfoglalhat, letolhatja a képernyőről az alsó LazyListet. Én úgy oldanám meg, hogy a kereső alatt csak egy LazyList legyen, valahogy így 

```kotlin
LazyColumn {
  items(currentlyAddedUsers) {UserRow(...)}
  if (notAddedFilteredUsers.isNotEmpty()) {
    item {divider}
    items(notAddedFilteredUsers) {UserRow(...)}
  }
}
```

Reply:
Illetve user választásra már van egy implementáció, bár lehet, hogy azt egy kicsit itt nyakatekert lenne használni. A UserAmountCard-ra gondolok (jelenleg az achievement_grant feature-ben van, ezt elszúrtam, ui.compose-ban kellene lennie és akkor meg lehetne találni). Itt a userre egy numerikus értéket lehet megadni, ha az érték maximuma 1 akkor nem plusz minusz gombok jelennek meg, hanem egy checkbox. Eköré lehete írni egy egyszerű wrappert ami eleve igaz hamis értékekkel dolgozik, és elnevezni UserBooleanCard-ra, így ugyanazt a UI implementációt kényelmesen lehetne erre is használni, és biztosan egységes lenne a felület. El tudom fogadni ha ezt túl nyakatekertnek gondolod, de nézz rá, és használd inkább ezt ha szerinted az ésszerűség határain belül van.

Reply:
ez sztem más use case. ha akarod, írhatsz rá wrappert, de én ezt nem látom kódduplikációs problémának, hogy ezért összefűzzem a kettőt